File: tuquanaoAI/lib/models/ai_evaluation.dart
class AiEvaluation {
  final String id;
  final String userId;
  final String topItemId;
  final String topItemName;
  final String bottomItemId;
  final String bottomItemName;
  final String? destination;
  final String? healthTag;
  final String? weatherSnapshot;
  final String evaluationText;
  final int? userRating; // 1–5
  final DateTime createdAt;

  const AiEvaluation({
    required this.id,
    required this.userId,
    required this.topItemId,
    required this.topItemName,
    required this.bottomItemId,
    required this.bottomItemName,
    this.destination,
    this.healthTag,
    this.weatherSnapshot,
    required this.evaluationText,
    this.userRating,
    required this.createdAt,
  });

  factory AiEvaluation.fromJson(Map<String, dynamic> json) => AiEvaluation(
    id:             json['id'] as String,
    userId:         json['userId'] as String,
    topItemId:      json['topItemId'] as String,
    topItemName:    json['topItemName'] as String,
    bottomItemId:   json['bottomItemId'] as String,
    bottomItemName: json['bottomItemName'] as String,
    destination:    json['destination'] as String?,
    healthTag:      json['healthTag'] as String?,
    weatherSnapshot:json['weatherSnapshot'] as String?,
    evaluationText: json['evaluationText'] as String,
    userRating:     json['userRating'] as int?,
    createdAt: DateTime.parse(json['createdAt'] as String),
  );

  Map<String, dynamic> toJson() => {
    'id':             id,
    'userId':         userId,
    'topItemId':      topItemId,
    'topItemName':    topItemName,
    'bottomItemId':   bottomItemId,
    'bottomItemName': bottomItemName,
    'destination':    destination,
    'healthTag':      healthTag,
    'weatherSnapshot':weatherSnapshot,
    'evaluationText': evaluationText,
    'userRating':     userRating,
    'createdAt':      createdAt.toIso8601String(),
  };

  AiEvaluation copyWith({int? userRating}) => AiEvaluation(
    id:             id,
    userId:         userId,
    topItemId:      topItemId,
    topItemName:    topItemName,
    bottomItemId:   bottomItemId,
    bottomItemName: bottomItemName,
    destination:    destination,
    healthTag:      healthTag,
    weatherSnapshot:weatherSnapshot,
    evaluationText: evaluationText,
    userRating:     userRating ?? this.userRating,
    createdAt:      createdAt,
  );
}

File: quan_ly_tu_do_API_one/quan_ly_tu_do_API/Models/Aievaluation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace quan_ly_tu_do_API.Models;

/// <summary>
/// Lưu lịch sử đánh giá trang phục từ AI (OutfitEvalViewModel → evaluate()).
/// Một bản ghi = user chọn áo + quần và nhận nhận xét từ AI.
/// </summary>
[Table("ai_evaluations")]
[Index("UserId", "CreatedAt", Name = "idx_ai_evaluations_user_date", IsDescending = new[] { false, true })]
public partial class AiEvaluation
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("user_id")]
    public Guid UserId { get; set; }

    // ── Trang phục được đánh giá ──────────────────────────────────────

    [Column("top_item_id")]
    public Guid TopItemId { get; set; }

    /// <summary>Snapshot tên áo (phòng trường hợp item bị xoá)</summary>
    [Column("top_item_name")]
    [StringLength(150)]
    public string TopItemName { get; set; } = null!;

    [Column("bottom_item_id")]
    public Guid BottomItemId { get; set; }

    /// <summary>Snapshot tên quần</summary>
    [Column("bottom_item_name")]
    [StringLength(150)]
    public string BottomItemName { get; set; } = null!;

    // ── Ngữ cảnh ──────────────────────────────────────────────────────

    [Column("destination")]
    [StringLength(150)]
    public string? Destination { get; set; }

    [Column("health_tag")]
    [StringLength(100)]
    public string? HealthTag { get; set; }

    [Column("weather_snapshot")]
    [StringLength(255)]
    public string? WeatherSnapshot { get; set; }

    // ── Kết quả AI ────────────────────────────────────────────────────

    /// <summary>Nội dung nhận xét AI trả về</summary>
    [Column("evaluation_text")]
    public string EvaluationText { get; set; } = null!;

    // ── Phản hồi user ─────────────────────────────────────────────────

    /// <summary>Điểm user tự chấm cho bộ đồ: 1–5 sao, null = chưa chấm</summary>
    [Column("user_rating")]
    public int? UserRating { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    // ── Navigation ────────────────────────────────────────────────────
    [ForeignKey("UserId")]
    public virtual User User { get; set; } = null!;

    [ForeignKey("TopItemId")]
    public virtual ClothingItem TopItem { get; set; } = null!;

    [ForeignKey("BottomItemId")]
    public virtual ClothingItem BottomItem { get; set; } = null!;
}